manifests/params.pp

```diff
@@ -197,14 +197,17 @@
   $client_certname            = $trusted['certname']
 
   # Set by the Foreman ENC
-  $puppetmaster               = getvar('puppetmaster')
+  $agent_server_hostname      = getvar('agent_server_hostname')
+  if $agent_server_hostname == '' {
+    $agent_server_hostname = getvar('puppetmaster')
+  }
```

Comment:
You can't reassign variables in Puppet. I also wouldn't expect it to return an empty string. This probably works:
```suggestion
  $agent_server_hostname = pick(getvar('agent_server_hostname'), getvar('puppetmaster'))
```

Then you also need something to modify the ENC output, which is set here:
https://github.com/theforeman/foreman_puppet/blob/5500ecd3d772b36fb80f71b2aff05822471167df/app/services/foreman_puppet/host_info_providers/puppet_info.rb#L38

For now I'd avoid this whole matter and keep it separate. So
```suggestion
  $agent_server_hostname = getvar('puppetmaster')
```

Then make an issue to tackle on its own. For example, we've talked about moving all Foreman details into its own hash. So you have a `foreman_enc` hash with all Foreman details, rather than a bunch at the top scope. That way you'll never introduce `agent_server_hostname`, but rather some other variable.

Reply:
That looks beyond my current limits.... yeah I'll revert these lines.